ui/src/app/tasks/task-definition/executions/task-definition-executions.component.ts

```diff
@@ -0,0 +1,144 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import { TasksService } from '../../tasks.service';
+import { SortParams } from '../../../shared/components/shared.interface';
+import { TaskExecution } from '../../model/task-execution';
+import { Page } from '../../../shared/model/page';
+import { TaskExecutionListParams } from '../../components/tasks.interface';
+import { Subject } from 'rxjs/Subject';
+
+/**
+ * Component that shows the executions of a Task
+ *
+ * @author Damien Vitrac
+ */
+@Component({
+  selector: 'app-task-definition-executions',
+  templateUrl: 'task-definition-executions.component.html',
+  styleUrls: ['../styles.scss'],
+})
+export class TaskDefinitionExecutionsComponent implements OnInit, OnDestroy {
+
+  /**
+   * Observable of page of task executions
+   */
+  executions$: Observable<any>;
+
+  /**
+   * Params Subject
+   */
+  params$: Subject<TaskExecutionListParams>;
+
+  /**
+   * Constructor
+   *
+   * @param {ActivatedRoute} route
+   * @param {Router} router
+   * @param {TasksService} tasksService
+   */
+  constructor(private route: ActivatedRoute,
+              private router: Router,
+              private tasksService: TasksService) {
+    this.params$ = new Subject<TaskExecutionListParams>();
+  }
+
+  /**
+   * Init component, call refresh method
+   */
+  ngOnInit() {
+    this.params$.subscribe((params: TaskExecutionListParams) => {
+      this.executions$ = this.tasksService.getTaskExecutions(params)
+        .mergeMap(
+          (page: Page<TaskExecution>) => Observable.of(params),
+          (page, params) => ({
+            page: page,
+            params: params
+          })
+        );
+    });
+    this.route.parent.params.subscribe((params: Params) => {
+      this.params$.next({
+        task: params.id,
+        sort: null,
+        order: null,
+        page: 0,
+        size: 30
+      });
+    });
+  }
+
+  /**
+   * On Destroy
+   */
+  ngOnDestroy(): void {
+    this.params$.complete();
+    this.params$.unsubscribe();
+  }
+
+  /**
+   * Refresh
+   * Create an observable which provides the required data
+   * @param {TaskExecutionListParams} params
+   */
+  refresh(params: TaskExecutionListParams) {
```

Comment:
I guess all the paging support can be abstracted away to a common module for reusability.